python/ray/autoscaler/_private/_azure/config.py

```diff
@@ -181,22 +186,97 @@ def _configure_resource_group(config):
 
 
 def _configure_key_pair(config):
+    """
+    Configure SSH keypair, generate one automatically if keypair is not specified and does not exist.
+    """
     ssh_user = config["auth"]["ssh_user"]
-    public_key = None
-    # search if the keys exist
-    for key_type in ["ssh_private_key", "ssh_public_key"]:
-        try:
-            key_path = Path(config["auth"][key_type]).expanduser()
-        except KeyError:
-            raise Exception("Config must define {}".format(key_type))
-        except TypeError:
-            raise Exception("Invalid config value for {}".format(key_type))
-
-        assert key_path.is_file(), "Could not find ssh key: {}".format(key_path)
-
-        if key_type == "ssh_public_key":
-            with open(key_path, "r") as f:
-                public_key = f.read()
+
+    # Check if user specified custom SSH key paths
+    user_specified_private_key = "ssh_private_key" in config["auth"]
+    user_specified_public_key = "ssh_public_key" in config["auth"]
+
+    # Validate that user specified both keys in config
+    if user_specified_private_key != user_specified_public_key:
+        if user_specified_private_key:
+            raise ValueError(
+                "ssh_private_key is specified but ssh_public_key is missing. "
+                "Both SSH key paths must be specified together, or omit both from your config to use auto-generated keys."
+            )
+        else:
+            raise ValueError(
+                "ssh_public_key is specified but ssh_private_key is missing. "
+                "Both SSH key paths must be specified together, or omit both from your config to use auto-generated keys."
+            )
+
+    private_key_path = Path(
+        config["auth"].get("ssh_private_key", "~/.ssh/id_rsa")
+    ).expanduser()
+    public_key_path = Path(
+        config["auth"].get("ssh_public_key", "~/.ssh/id_rsa.pub")
+    ).expanduser()
+
+    # If user specified both keys in config, validate they exist
+    if user_specified_private_key and user_specified_public_key:
+        nonexistent_keys = []
+        if user_specified_private_key and not private_key_path.is_file():
+            nonexistent_keys.append(f"ssh_private_key: {private_key_path}")
+        if user_specified_public_key and not public_key_path.is_file():
+            nonexistent_keys.append(f"ssh_public_key: {public_key_path}")
+
+        if nonexistent_keys:
+            raise ValueError(
+                "Specified SSH key files do not exist: {}. "
```

Comment:
nit `"User-specified SSH keys files do not exist: {}. "`

Reply:
or maybe `"SSH key files from config do not exist: {}. "` to match the below info message